src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,48 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import mate.academy.exception.NoUserInCacheException;
+import mate.academy.exception.RequestProcessingException;
 
 public class AsyncRequestProcessor {
+    private static Map<String, UserData> cache = new ConcurrentHashMap<>();
```

Comment:
Using a static cache in a non-static context can lead to issues if multiple instances of `AsyncRequestProcessor` are created. Consider removing the static keyword if the cache is intended to be unique per instance or ensuring thread safety and clear lifecycle management if it's intended to be shared.